### PR TITLE
Removed `cmd/otelcol`, removed Make targets and github actions workflow

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:** 
This PR removes the `cmd/otelcol` directory and makes the appropriate changes to the Makefile to remove targets that depend on `cmd/otelcol`. This PR also removes some Github workflows in `.github/workflows` that reference removed make targets. This PR is the second of 2 PRs, which both together are equivalent to PR #3992. This previous PR is split into two to first ensure that the `collector-builder` works as intended before we delete `cmd/otelcol`.

**Link to tracking Issue:** #3928

### CODE REVIEW

**Comment by erichsueh3** [comment]
> Looks like the tests are failing still

@codeboten Interesting, I'm really still not sure why `build-and-test/unittest` is still failing; it passed after the first commit in this PR, and failed after the second commit (which only added back in the `make run` target that currently doesn't do anything, [see here](https://github.com/open-telemetry/opentelemetry-collector/pull/4200/commits/c173dce92191aaf70a46d05a86f004cb10004e19).

**Review by carlosalberto — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by erichsueh3** [comment]
EasyCLA test hanging, might need to be re-run

File: Makefile [hunk 1]
```diff
@@ -211,74 +202,6 @@ delete-tag:
 docker-otelcol:
```
Comment (jpkrohling) [inline Makefile:202]: Doesn't this target require the binary that we are not building anymore?

File: Makefile [hunk 2]
```diff
@@ -140,19 +140,10 @@ install-tools:
...
-
-.PHONY: otelcol-unstable
-otelcol-unstable:
-	go generate ./...
-	$(MAKE) build-binary-internal-unstable
-
 .PHONY: run
-run:
-	GO111MODULE=on go run --race ./cmd/otelcol/... --config ${RUN_CONFIG} ${RUN_ARGS}
+run: build-binary-cmd-otelcol
```
Comment (jpkrohling) [inline Makefile:144]: This currently fails with:

```
$ make run
make: *** No rule to make target 'build-binary-cmd-otelcol', needed by 'run'.  Stop.
```

Perhaps you need to rebase?

File: Makefile [hunk 3]
```diff
@@ -211,74 +202,6 @@ delete-tag:
 docker-otelcol:
 	COMPONENT=otelcol $(MAKE) docker-component
```
Comment (jpkrohling) [inline Makefile:203]: The target `docker-component` can probably be removed as well, as `docker-otelcol` is the only one calling it.

**Review by jpkrohling — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (erichsueh3) [reply Makefile]: Removed

File: Makefile (hunk 3 above)
Comment (erichsueh3) [reply Makefile]: Removed

File: Makefile (hunk 2 above)
Comment (erichsueh3) [reply Makefile:144]: For some reason even after rebasing the `build-binary-cmd-otelcol` rule was deleted in this PR; added it back in however and `make run` runs properly.

**Comment by erichsueh3** [comment]
Is a `CHANGELOG` entry required for this?

**Comment by jpkrohling** [comment]
> Is a CHANGELOG entry required for this?

Not sure -- I would assume that the changelog is mostly for consumers of the collector, not people developing the collector.

**Review by jpkrohling — APPROVED** [review]

**Comment by alolita** [comment]
@carlosalberto @Aneurysm9 can you please re-review to merge. Ty!

**Pull request merged**